File: Insertion/insert.c
#include <stdio.h>

#define MAX 100

int main() {

    //declaration
    int arr[MAX];
    int n, i;
    int index_pos;
    int element;
    int flag = 1;


    printf("Enter the number of elements (up to %d): ", MAX);
    scanf("%d", &n);

    printf("Enter the elements:\n");
    for (i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    while(flag == 1){
        printf("Enter the index where the element is to be inserted: ");
        scanf("%d", &index_pos);

        if( index_pos >= 0 && index_pos <= n-1){
            flag = 0;
        } else {
            printf("The index is out of bound, \n");
            flag = 2;
        }
   }    
    printf("Enter the element to be inserted: ");
    scanf("%d", &element);

    for (i = n - 1; i >= index_pos; i--) {
        arr[i + 1] = arr[i];
    }

    arr[index_pos] = element;
    n++; 

    printf("The elements entered are: ");
    for (i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
 
 
    return 0;

}

File: Insertion/insert.java
import java.util.Scanner;

public class insert {
    public static void main(String[] args) {
        final int MAX_SIZE = 100;
        int[] arr = new int[MAX_SIZE];
        int n, i;
        int indexPos;
        int element;

        try (Scanner scanner = new Scanner(System.in)) {
            System.out.printf("Enter the number of elements (up to %d): ", MAX_SIZE);
            n = scanner.nextInt();

            System.out.println("Enter the elements:");
            for (i = 0; i < n; i++) {
                arr[i] = scanner.nextInt();
            }

            System.out.print("Enter the index where the element is to be inserted: ");
            indexPos = scanner.nextInt();

            System.out.print("Enter the element to be inserted: ");
            element = scanner.nextInt();

            // Shift elements to the right to make space for the inserted element
            for (i = n - 1; i >= indexPos; i--) {
                arr[i + 1] = arr[i];
            }

            arr[indexPos] = element;
            n++; // Increment the size of the array

            System.out.print("The elements entered are: ");
            for (i = 0; i < n; i++) {
                System.out.print(arr[i] + " ");
            }
            System.out.println();

            scanner.close();
        }
    }
}
